Add etymology and sales-pitch to doc front-page

loenard97 opened the pull request:
#### Description

Fixes #3484
Fixes #2442

 - Added an extended tagline on the front-page of the documentation website, which includes the origin of the name "Yew" according to Issue #2442
  - Added a short sales pitch on what makes Yew special compared to other frameworks. I'm not that familiar with pros and cons of different frameworks, so more ideas on what sets Yew apart would be an improvement, I think. Any thoughts?

#### Checklist
- [x] I have reviewed my own code
- [ ] I have added tests

hamza1311 commented:
I feel like this is a lot of text that is thrown at the reader:
![image](https://github.com/yewstack/yew/assets/47357913/750391f3-0b19-47d0-9668-5b022e85df7e)

> Yew (pronounced yo͞o with a soft y, like in "you") is named after the giant humanoid yew tree from the movie "A Monster Calls", who said it best: "Only Yew can help you"
This part also feels irrelevant as the 2nd of front page

loenard97 commented:
Agreed. I tried putting the text at different positions and this was the best I could come up with for the moment. Maybe it makes more sense as an unordered list? Then it won't look as cluttered. Or maybe it could be at the top of the Getting Started page? I'm not really sure.

loenard97 commented:
I experimented a bit more with it, but couldn't find a good place for where to put the text. I moved it to the README.md instead like the original issue suggested.

In the website/README.md I also added a short section about how to run the formatter, because I didn't know about that on my first commit.
